controllers/namespace_controller.go

```diff
@@ -351,7 +374,38 @@ func (r *NamespaceReconciler) reconcileTemplateNamespace(ctx context.Context, ns
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *NamespaceReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	subNSHandler := func(o client.Object, q workqueue.RateLimitingInterface) {
+		q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
+			Name: o.GetName(),
+		}})
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&corev1.Namespace{}).
+		Watches(&source.Kind{Type: &accuratev1.SubNamespace{}}, handler.Funcs{
+			CreateFunc: func(ev event.CreateEvent, q workqueue.RateLimitingInterface) {
+				subNSHandler(ev.Object, q)
+			},
+			UpdateFunc: func(ev event.UpdateEvent, q workqueue.RateLimitingInterface) {
+				if ev.ObjectNew.GetDeletionTimestamp() != nil {
+					return
+				}
+				subNSHandler(ev.ObjectOld, q)
+			},
+			DeleteFunc: func(ev event.DeleteEvent, q workqueue.RateLimitingInterface) {
+				subNSHandler(ev.Object, q)
```

Comment:
This is not necessary, because when SubNamespace is deleted, the target Namespace will be also deleted.